arrays_lab.js

```diff
@@ -0,0 +1,299 @@
+// (The first element has an index of 0). Use negative numbers to select
+// from the end of an array. If omitted, it acts like "0"
+// end	Optional. An integer that specifies where to end the selection.
+// If omitted, all elements from the start position and to the end of the array will be selected. Use negative numbers to select from the end of an array
+//
+//
+// //no The splice() method adds/removes items to/from an array,
+// //and returns the removed item(s).Like below:
+//
+// //var fruits = ["Banana", "Orange", "Apple", "Mango"];
+// //fruits.splice(2, 0, "Lemon", "Kiwi");
+//
+// //Note: This method changes the original array.
+//
+// //At position 2, add the new items, and remove 1 item:
+// //var fruits = ["Banana", "Orange", "Apple", "Mango"];
+// //fruits.splice(2, 1, "Lemon", "Kiwi");
+// //At position 2, add the new items, and remove 1 item:
+// //var fruits = ["Banana", "Orange", "Apple", "Mango"];
+// //fruits.splice(2, 1, "Lemon", "Kiwi");
+//console.log("_____________________________________________________________")
+// //3. Add "mango" to the end fruits array ["strawberry", "banana"].
+let fruits = ["strawberry", "banana"]
+fruits.push("mango")
+console.log(fruits)
+
+//console.log("_____________________________________________________________")
+// //4. Add "blueberry" to the front of the same fruits array.
+fruits.unshift("blueberry")
+console.log(fruits)
+//console.log("_____________________________________________________________")
+// //5. Remove the last element of the fruits array.
+fruits.pop()
+console.log(fruits)
+//console.log("_____________________________________________________________")
+
+// //6. console.log the length of the fruits array.
+console.log(fruits.length)
+
+//console.log("_____________________________________________________________")
+// //7. Remove the first element of the fruits array.
+
+fruits.shift()
+console.log(fruits)
+//console.log("_____________________________________________________________")
+// //8. Join all the elements in the fruits array with a '$'.
+
+console.log(fruits.join(" $ "))
+//console.log("_____________________________________________________________")
+
+//9. Given two arrays, write code that logs the larger array.
+//If the arrays are the same length, log "They are the same size"
+
+
+let firstArr = [1, 2, 3];
+let secondArr = [1, 5, 2, 4];
+if (firstArr.length > secondArr.length) {
+  console.log(firstArr);
+} else if (firstArr.length < secondArr.length){
+  console.log(secondArr);
+} else {
+  console.log("They are the same size")
+}
+
+//console.log(----------------------------------------------------------------)
+//10)
+//Given an array, log its middle element.
+//If the middle of the array is between two elements, log both of them.
+
+//let hasExactMiddleArr = [1,5,3] -// log 5
+
+//let hasNoExactMiddleArr = [1,4,6,9]-// log 4,6
+
+let numArray = [2,3,4,5,6];
+let midArr = Math.floor(numArray.length -1 / 2);
```

Comment:
I see some good ideas here like defining index number for middle array. Also (numArray.length - 1) is not correct. One other caveat for this code is that this code wouldn't work with even length array. How can we solve this? hint: Even number array has two middle numbers, thus numArray[midArr], numArray[midArr - 1].